pkg/traces/config.go

```diff
@@ -354,11 +408,39 @@ func (c *InstanceConfig) exporters() (map[string]interface{}, error) {
 			return nil, err
 		}
 		exporterName = fmt.Sprintf("%s/%d", exporterName, i)
+		if remoteWriteConfig.Oauth2 != nil {
```

Comment:
this makes more sense to me in the `exporter()` function since we build the rest of the config there.

I believe it's here b/c the `exporterName` is here? Perhaps we could adjust `exporter()` to take its own name and use it?

Oof, it also seems odd that `getExporterName()` doesn't take `i`. Tacking it on afterwards seems error prone.

Reply:
Good point. I agree.
66315ced